### DISCUSSION TITLE
New long variables making my build size larger

### DISCUSSION THREAD

**Discussion by Steffan153** [tailwindlabs/tailwindcss#13199] [General]
In v4, my build CSS no longer has all the `/var(--tw-bg-opacity)` stuff on every color I use, which is nice. I was hoping my build size would now be much smaller. But in the end, it ends up a lot larger because of all the new theme variables. Could there be a setting or something to not put everything in `--color-violet-50` etc in `:root`, but just use them inline in the CSS rules? Or maybe minify the variable names?

(I like to have the smallest build sizes possible)

**Comment by adamwathan** [comment]
Hey! Still finalizing some details but we do support this currently with this `reference` option on your theme, try this:

```diff
- @import "tailwindcss";
+ @layer theme, base, components, utilities;
+ @import 'tailwindcss/theme' reference;
+ @import 'tailwindcss/preflight' layer(base);
+ @import 'tailwindcss/utilities' layer(utilities);
```

This tells Tailwind to process your `@theme` as a "reference", so it's used internally to know what values exist but isn't included in the final CSS.

You can do the same thing directly on a `@theme` rule like this:

```css
@theme reference {
  ...
}
```

**Comment by Steffan153** [comment]
Awesome! My build size is now 27KB with v4, compared to 31KB with v3. Thanks!
